test/UnitTests/MSTest.CoreAdapter.Unit.Tests/Discovery/UnitTestDiscovererTests.cs

```diff
@@ -165,16 +173,39 @@ public void SendTestCasesShouldSendTestCasesWithNaigationData()
             var testElements = new List<UnitTestElement> { test };
 
             this.SetupNavigation(source, test, test.TestMethod.FullClassName, test.TestMethod.Name);
+            this.mockRunSettings.Setup(rs => rs.SettingsXml).Returns(settingsXml);
 
             // Act
-            this.unitTestDiscoverer.SendTestCases(source, testElements, this.mockTestCaseDiscoverySink.Object);
+            this.unitTestDiscoverer.SendTestCases(source, testElements, this.mockTestCaseDiscoverySink.Object, this.mockRunSettings.Object);
 
             // Assert
-            this.mockTestCaseDiscoverySink.Verify(ds => ds.SendTestCase(It.Is<TestCase>(tc => tc.FullyQualifiedName == "C.M")), Times.Once);
+            this.mockTestCaseDiscoverySink.Verify(ds => ds.SendTestCase(It.Is<TestCase>(tc => tc.LineNumber == -1)), Times.Once);
+            this.mockTestCaseDiscoverySink.Verify(ds => ds.SendTestCase(It.Is<TestCase>(tc => tc.CodeFilePath == null)), Times.Once);
         }
 
         [TestMethodV1]
-        public void SendTestCasesShouldSendTestCasesWithNaigationDataWhenDeclaredClassFullNameIsNonNull()
+        public void SendTestCasesShouldSendTestCasesWithNavigationData()
+        {
+            var source = "DummyAssembly.dll";
```

Comment:
Source is DummyAssembly in most scenarios, should it be moved as a const in the class? Can the test and testElements be instance variables, they are same as well?